core/src/main/java/io/grpc/internal/TransportSet2.java

```diff
@@ -356,115 +358,21 @@ public String getLogId() {
-    return new ClientCallImpl<RequestT, ResponseT>(methodDescriptor,
-        new SerializingExecutor(appExecutor), callOptions, StatsTraceContext.NOOP,
-        new ClientTransportProvider() {
-          @Override
-          public ClientTransport get(CallOptions callOptions) {
-            return obtainActiveTransport();
-          }
-        },
-        scheduledExecutor);
-  }
-
-  @Override
-  public boolean isShutdown() {
+  @VisibleForTesting
+  ConnectivityState getState() {
     synchronized (lock) {
-      return shutdown;
-    }
-  }
-
-  @Override
-  public boolean isTerminated() {
-    return terminatedLatch.getCount() == 0;
-  }
-
-  @Override
-  public boolean awaitTermination(long timeout, TimeUnit unit) throws InterruptedException {
-    return terminatedLatch.await(timeout, unit);
-  }
-
-  @Override
-  public String authority() {
-    return authority;
-  }
-
-  @Override
-  public ConnectivityState getState(boolean requestConnection) {
-    if (requestConnection) {
-      boolean connect = false;
-      synchronized (lock) {
-        connect = stateManager.getState() == ConnectivityState.IDLE;
-      }
-      if (connect) {
-        obtainActiveTransport();
-      }
-    }
-    synchronized (lock) {
-      return stateManager.getState();
-    }
-  }
-
-  @Override
-  public void notifyWhenStateChanged(ConnectivityState source, Runnable callback) {
-    synchronized (lock) {
-      stateManager.notifyWhenStateChanged(callback, appExecutor, source);
-    }
-  }
-
-  /** Shared base for both delayed and real transports. */
-  private class BaseTransportListener implements ManagedClientTransport.Listener {
-    protected final ManagedClientTransport transport;
-
-    public BaseTransportListener(ManagedClientTransport transport) {
-      this.transport = transport;
-    }
-
-    @Override
-    public void transportReady() {}
-
-    @Override
-    public void transportInUse(boolean inUse) {
-      inUseStateAggregator.updateObjectInUse(transport, inUse);
-    }
-
-    @Override
-    public void transportShutdown(Status status) {}
-
-    @Override
-    public void transportTerminated() {
-      boolean runCallback = false;
-      inUseStateAggregator.updateObjectInUse(transport, false);
-      synchronized (lock) {
-        transports.remove(transport);
-        if (shutdown && transports.isEmpty()) {
-          if (log.isLoggable(Level.FINE)) {
-            log.log(Level.FINE, "[{0}] Terminated in transportTerminated()", getLogId());
-          }
-          terminatedLatch.countDown();
-          runCallback = true;
-          cancelReconnectTask();
-        }
-      }
-      if (runCallback) {
-        callback.onTerminated(TransportSet.this);
-      }
+      return state.getState();
     }
   }
 
   /** Listener for real transports. */
-  private class TransportListener extends BaseTransportListener {
+  private class TransportListener implements ManagedClientTransport.Listener {
+    final ManagedClientTransport transport;
```

Comment:
It doesn't matter since the class is private, I suppose? For the sake of consistency and conciseness, I would like to remove all the access modifiers.